### PR TITLE
remove default kubeconfig from kubeadm token

### LINKED ISSUE #198: `kubeadm token` commands should look in standard places for kubeconfig
Currently the `kubeadm token` commands assumes it can find it's kubeconfig at `/etc/kubernetes/admin.conf`.  Instead, we should probably also look for the kubeconfig in the standard places and respect the standard env variables so that this can be used like `kubectl`.

Another option would be to move/duplicate the token management methods into `kubectl` directly as they can be managed off cluster.

### PR DESCRIPTION
**Pull request by xilabao**
The problem is similar to https://github.com/kubernetes/kubernetes/pull/42970
fixes https://github.com/kubernetes/kubeadm/issues/198

### CODE REVIEW

**Review by pires — APPROVED** [review]
/lgtm

**Comment by xilabao** [comment]
This is bug fix. Should we merge it to 1.6?

**Comment by xilabao** [comment]
ping @pires

**Comment by pires** [comment]
@jbeda @luxas what say you?

File: cmd/kubeadm/app/util/kubeconfig/kubeconfig.go [hunk 1]
```diff
@@ -68,6 +69,10 @@ func CreateWithToken(serverURL, clusterName, userName string, caCert []byte, tok
 
 // ClientSetFromFile returns a ready-to-use client from a KubeConfig file
 func ClientSetFromFile(path string) (*clientset.Clientset, error) {
+	if len(path) == 0 {
+		// get the default kubeconfig from env
+		path = os.Getenv("KUBECONFIG")
```
Comment (luxas) [inline cmd/kubeadm/app/util/kubeconfig/kubeconfig.go]: What happens if `KUBECONFIG` is `""`? Then it won't pass any path, which might be bad. 
Have you checked that scenario?

Comment (xilabao) [inline cmd/kubeadm/app/util/kubeconfig/kubeconfig.go]: if `KUBECONFIG` is `""`, and `clientcmd.LoadFromFile` will return the error.  People may use `--kubeconfig=invalidpath` too.

**Comment by xilabao** [comment]
@luxas PTAL

**Comment by luxas** [comment]
Sorry, but I believe this will break kubeadm.
If no `--kubeconfig` is passed and `KUBECONFIG` isn't set, it seems like `""` will be passed; which I think will error out. Also, it should point to `/etc/kubernetes/admin.conf` by default.

cc @kubernetes/sig-cli-pr-reviews @jbeda can you think of the best solution here?
Can we reuse some of the kubectl logic here?

**Comment by liggitt** [comment]
why is kubeadm using a non-standard default path for looking up the kubeconfig file? I would have expected to either use the standard paths or require the user to specify `--kubeconfig`

**Comment by luxas** [comment]
@liggitt in order to not clobber that much with the standard kubeconfig file; which in most cases would be under `/root/.kube/...` anyway since the user has to be root to run `kubeadm init`.
The admin.conf file has cluster-admin privs and should therefore be protected as well; we didn't want it to be written to a random user's `~/.kube/config` file

**Comment by ericchiang** [comment]
> we didn't want it to be written to a random user's ~/.kube/config file

It seems like you would want to avoid using the KUBECONFIG environment variable too then.

**Comment by xilabao** [comment]
@luxas `kubeadm init` guide people to run the command:
```
To start using your cluster, you need to run (as a regular user):

  sudo cp /etc/kubernetes/admin.conf $HOME/
  sudo chown $(id -u):$(id -g) $HOME/admin.conf
  export KUBECONFIG=$HOME/admin.conf
```
I think it's necessary to support the KUBECONFIG environment.

**Pull request closed without merging**